aries_cloudagent/protocols/didexchange/v1_0/manager.py

```diff
@@ -740,7 +746,9 @@ async def accept_response(
             raise DIDXManagerError("No DIDDoc attached; cannot connect to public DID")
         async with self.profile.session() as session:
             wallet = session.inject(BaseWallet)
-            conn_did_doc = await self.verify_diddoc(wallet, response.did_doc_attach)
+            conn_did_doc = await self.verify_diddoc(
+                wallet, response.did_doc_attach, conn_rec.invitation_key
```

Comment:
This checks whether the did document is signed with the first verification method (code above in this file assign conn_rec.invitation_key to first verificationMethod). I think this should be signed using an `authentication` key. Also I'm not sure which key we should check (the RFC is not very clear), but it could be signed with the second (or third) `authentication` key I guess.

Maybe @andrewwhitehead has some insights on this

Reply:
Based upon the clarification received during Aries WG 2022/03/23, updated [verify method](https://github.com/hyperledger/aries-cloudagent-python/blob/790ab5e4034f2017efb45e71a6d183323a4042dd/aries_cloudagent/messaging/decorators/attach_decorator.py#L417) and now using `jws.header.kid` to verify signature.